interview-questions.md

```diff
@@ -5,46 +5,91 @@ First, without external resources. Challenge yourself to answer from memory.
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-1. As a developer, I am creating an API with a model called Animal that has_many Sightings, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
+1. As a developer, I am creating an API with a model called Animal that has_many Sightings, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key?
+    Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
+  Your answer: Because it is the belongs_to association that is missing, you would create and run a migration called AddBelongsToAnimalToSighting that changes the sighting table to have the
+                `t.belongs_to :animal` association. Rails knows what to do with this, and adds an integer value of animal_id to the sighting table. This animal_id integer is the foreign key and
+                is part of the Sighting model. If you also forgot to add the other half of the association, all that is needed is to add a line to the Animal model file that states `has_many :sightings`
+                The above migration will not execute if the Animal model is missing this association declaration.
 
   Researched answer:
+              After doing some research, apparently this solution is relatively new. I actually couldn't find a result on the first page of search results that mentioned this approach. The most common method I saw was
+              to create a migration that uses `add_foreign_key <model>_id` for the child table. These solutions were all being posted in relation to releases < Rails 5.0. After seeing all these results I began to
+              doubt myself(I was confident that my method would work when creating the table but hadn't actually tested it for a migration for an already extant table) so I created a new model in my Wildlife Tracker
+              with no attributes and then ran a separate migration to add the belongs_to declaration. It worked as expected, so I have concluded that this must be a technique that wasn't always a part of Rails. I do
+              know it exists in Rails 5 so I would assume that is when it was added.
```

Comment:
Yea, Knowing that rails can do both I wonder when the convention we teach in class came about too